src/partition/repart.c

```diff
@@ -6963,6 +7071,37 @@ static int parse_argv(int argc, char *argv[]) {
                         *p = gpt_partition_type_override_architecture(*p, arg_architecture);
         }
 
+        if (arg_signing_engine && !arg_private_key_path)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "--private-key must be specified when --signing-engine is used.");
```

Comment:
as per coding style we generally try to suffix options that expect arguments with `=`